spring-kafka/src/main/java/org/springframework/kafka/support/converter/Jackson2JavaKeyTypeMapper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support.converter;
+
+/**
+ * Jackson 2 type mapper for record keys.
+ *
+ * @author Elliot Kennedy
+ *
+ * @since 2.1
+ */
+public class Jackson2JavaKeyTypeMapper extends DefaultJackson2JavaTypeMapper {
```

Comment:
Well, actually that's not what we have talked about.

The idea is like make the types properties *configurable* in the `DefaultJackson2JavaTypeMapper`.

Introducing this new class you are mixing concerns and say explicitly what it is about and at the same time narrow its functionality.

So, my point is to provide setters for those field names.

This way we still have the same `DefaultJackson2JavaTypeMapper`, but now it is more flexible.

Does it make sense to you?

Reply:
Thanks @artembilan - I'm still not sure about the isKey constructor argument. Should it configure the headers as I have done or is that overkill?

Please take a look